packages/board-server/src/server/common.ts

```diff
@@ -3,16 +3,33 @@ import type { IncomingMessage, ServerResponse } from "http";
 import { dirname, extname, resolve } from "path";
 import { fileURLToPath } from "url";
 import type { ViteDevServer } from "vite";
+import { existsSync } from 'fs';
 
 import type { ServerConfig } from "./config.js";
 import { notFound, serverError } from "./errors.js";
 import type { PageMetadata } from "./types.js";
 import { createReadStream } from "fs";
 
 const MODULE_PATH = dirname(fileURLToPath(import.meta.url));
-const ROOT_PATH = resolve(MODULE_PATH, "../..");
+
+function findRootPath() {
```

Comment:
So I suspect this is a function of my inexperience with JS/TS/Vite. I found that when running from the tests, the root path that was getting resolved was always one level wrong, and this meant that the root path included `/dist` on the end, but so did the `client_path`.

So in a sense, I've bailed out with this more sophisticated pathing approach. If somebody else has more insight I would welcome it. Perhaps @jamesjacko or @Mearman might have a better idea why I was facing this issue?

Reply:
I have a feeling that the issue may be because the value is declared as a `const`. If you compute the "`base path`" at the apex of the execution and pass it down via the `makeRouter`, it might resolve it.

Reply:
Ooh, thank you for the suggestion @Mearman, let me have a play with that idea..

Reply:
I've incorporated your suggestion @Mearman in [f4d514c](https://github.com/breadboard-ai/breadboard/pull/2933/commits/f4d514c31501d21fd5ca79a499609a50f40d08db). Thank you for that, I think I finally understand the problem. When starting the server from a test context, the module path is `dist/tests/integration`, so I need to ascend three levels rather than two. I have added a parameter in the server start to override the root path from tests - putting this in the server config which is already passed around, this leaves the main logic in common.ts untouched.